src/main/java/org/breskul/connectivity/datasource/DataSourcePropertyResolver.java

```diff
@@ -0,0 +1,82 @@
+package org.breskul.connectivity.datasource;
+
+import lombok.extern.slf4j.Slf4j;
+import org.breskul.exception.BoboException;
+import org.breskul.exception.ExceptionMessage;
+
+import java.io.*;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.breskul.util.Constants.*;
+
+@Slf4j
+public class DataSourcePropertyResolver {
+    private Map<String, String> valueToPropertyName;
+
+    public DataSourcePropertyResolver() {
+        scanFile(DEFAULT_PROPERTIES_FILE_NAME);
+        initializeDataSource();
+    }
+
+    public DataSourcePropertyResolver(String fileName) {
+        scanFile(fileName);
+        initializeDataSource();
+    }
+
+    private void initializeDataSource() {
+        if (Objects.nonNull(valueToPropertyName) && !valueToPropertyName.isEmpty()) {
+            String url = valueToPropertyName.get(URL);
+            String username = valueToPropertyName.get(USERNAME);
+            String password = valueToPropertyName.get(PASSWORD);
+
+            if (isEmpty(url) || isEmpty(username) || isEmpty(password)) {
+                log.warn("Could not find one of the properties in application.properties file: {}, {}, {}. DataSource has not been initialized.", URL, USERNAME, PASSWORD);
+                return;
+            }
+            FileConfigurableDataSource.getInstance(url, username, password);
+        }
+    }
+
+    private void scanFile(String fileName) {
```

Comment:
I would suggest reading properties file like (This is for correct species symbols handling and mirroring): 
Properties prop = new Properties();
ClassLoader loader = Thread.currentThread().getContextClassLoader();           
InputStream stream = loader.getResourceAsStream("/myProp.properties");
prop.load(stream);